### PR TITLE
Add Scala extension

### PR DESCRIPTION
**Pull request by hmemcpy**
This PR adds support for the Scala language, based on Metals (Scala LSP).

### CODE REVIEW

File: extensions.toml [hunk 1]
```diff
@@ -271,6 +271,10 @@ version = "0.0.1"
 submodule = "extensions/r"
 version = "0.0.1"
 
+[scala]
+submodule = "extensions/scala"
```
Comment (hmemcpy) [inline extensions.toml:275]: Ah! My bad, thanks a lot!

**Comment by maxdeviant** [comment]
@hmemcpy Thanks for adding this!

**Comment by maxdeviant** [comment]
> Thank you for merging! So this should just appear in the extension gallery now? :)

Yep, it's there now!

Screenshot 2024-03-28 at 12 32 24 PM

You do need to be on the Preview channel (Zed v0.129.x) to use it, as that is the version that supports the new language server extensions. Once v0.129.x lands on Stable next Wednesday then everyone will be able to see it.

**Comment by hmemcpy** [comment]
Amazing! Thank you so much for your help making contributing to Zed a pleasure :)

**Comment by bishabosha** [comment]
Awesome! Thanks everyone 🙏

**Comment by samgj18** [comment]
image
Am I looking at the wrong place?

**Comment by hmemcpy** [comment]
@samgj18 it's only currently available for the [Preview release of Zed](https://zed.dev/releases/preview) (0.129.1), it should land in Stable hopefully next week.

**Comment by samgj18** [comment]
I see. Thanks a lot, @hmemcpy!

**Pull request merged**